api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -92,7 +92,7 @@ pub enum KubernetesInfrastructureError {
 impl KubernetesInfrastructure {
     pub fn new(
         cluster_endpoint: Url,
-        cluster_ca: Option<Certificate>,
+        cluster_ca: Option<Vec<Vec<u8>>>,
```

Comment:
I would rather move the parsing of the Der into the configuration object instead of passing bytes

Reply:
Do you mean changing the signature to `Option<Vec<Der>>`?

I chose to use `Vec<Vec<u8>>` to avoid needing `kube`-specifics (`kube::config::Der`) in the config and because `kube` 0.49 and newer actually expect bytes as input.

I can go ahead though and make the change!

Reply:
Okay, I see your point.

But, If the code passes just the bytes than the validation if the user has passed a path to valid certificate will happen very late at runtime (ignore the inner workings of `read_ca_file` for now). That might be confusing because he or she has just pointed the configuration to a file that does not contain a valid certificate. Therefore, I would be better if `read_ca_file` ensures that the file contains a valid certificate to inform the user that there is something as soon as possible.

There is now `rustls-pemfile` in use within `read_ca_file` but it is very young and has three commit in it (not happy with that). Therefore, I would recommend to use [`openssl's X509`](https://docs.rs/openssl/0.10.33/openssl/x509/struct.X509.html) because it is a dependency anyway and it gives the X509 type that you can use here. Based on this type you can convert it [`to_der`](https://docs.rs/openssl/0.10.33/openssl/x509/struct.X509.html#method.to_der)

Reply:
Got it!

fwiw, `rustls-pemfile` is not *completely* without a history, as it was just [spun out of rustls](https://github.com/ctz/rustls#release-history), but OpenSSL is already there, and you just did the hard work for me by looking up the right methods ;)